tests/test_cfg.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+from project.cfg import process_cnf_from_file
+
+
+def test_wrong_file():
+    path_not_exists = "tests/data/cfg/emp"
+    path_not_txt = "tests/data/cfg/empty"
+    path_to_empty = "tests/data/cfg/empty.txt"
+
+    with pytest.raises(OSError):
+        process_cnf_from_file(path_not_exists)
+    with pytest.raises(OSError):
+        process_cnf_from_file(path_not_txt)
+    with pytest.raises(OSError):
+        process_cnf_from_file(path_to_empty)
+
+
+def test_wrong_text():
+    with pytest.raises(ValueError):
+        process_cnf_from_file("tests/data/cfg/incorrect_grammar.txt")
+
+
+@pytest.mark.parametrize(
+    "filename, axiom",
+    [("epsilon.txt", "epsilon"), ("grammar.txt", "S"), ("random.txt", "Hello")],
```

Comment:
Кажется тут всё-таки должно быть "E"
```suggestion
    [("epsilon.txt", "E"), ("grammar.txt", "S"), ("random.txt", "Hello")],
```

Reply:
И, если вы посмотрите на `cnf.productions`, то получите `set()`, хотя в ОНФХ было бы `E -> epsilon`